socorro/unittest/external/postgresql/test_adi.py

```diff
@@ -0,0 +1,143 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import datetime
+
+from nose.plugins.attrib import attr
+from nose.tools import eq_, assert_raises
+
+from socorro.external.postgresql.adi import ADI
+from socorro.external import MissingArgumentError
+
+from unittestbase import PostgreSQLTestCase
+
+
+@attr(integration='postgres')  # for nosetests
+class IntegrationTestADI(PostgreSQLTestCase):
+
+    def setUp(self):
+        """Set up this test class by populating the reports table with fake
+        data. """
+        super(IntegrationTestADI, self).setUp()
+        self._truncate()
+        cursor = self.connection.cursor()
+
+        self.date = datetime.datetime(2015, 7, 1)
+        yesterday = self.date.replace(hour=0) - datetime.timedelta(hours=24)
```

Comment:
Why do you need ``.replace(hour=0)``? It should already be ``0`` since you didn't pass a value for the ``hour`` param when you created it. And also, why not just make it a ``datetime.date`` object?